server/Network/RequestHandler.ts

```diff
@@ -137,6 +127,37 @@ app.post('/connectWaiter', (_req, res) => {
 	res.send(waiter.connectWaiter());
 });
 
-app.listen(PORT, () => {
+io.on('connection', function (socket: socketio.Socket) {
+	console.log('a user connected');
+	sockets.push(socket);
+	socket.on('message', (message: any) => {
+		switch (message['kind']) {
+			case 'updateGuestLocation': {
+				guest.updateLocationGuest(
+					message['location'],
+					socket.handshake.auth['token']
+				);
+				break;
+			}
+			case 'updateWaiterLocation': {
+				waiter.updateLocationWaiter(
+					socket.handshake.auth['token'],
+					message['map'],
+					message['location']
+				);
+				break;
+			}
+			default: {
+				socket.emit('message', {
+					kind: 'error',
+					content: 'This kind of massage is not supported',
+				});
+				break;
+			}
+		}
+	});
+});
```

Comment:
Why use message['kind'], a cleaner approach would be to simply use different event names for each type of event, such as 
`socket.on("updateWaiterLocation", callback)`

https://socket.io/docs/v4/server-api/#socketoneventname-callback

Reply:
oops, I checked if there's something like that but couldn't find that.